spec/api.spec.js

```diff
@@ -308,14 +309,21 @@ describe('API', function () {
         })
       }
 
-      let tempSpec = Object.assign({}, specData, options, { onCluster })
-      clusterInfo = Object.assign({}, clusterDetail, { specData: tempSpec })
+      clusterInfo = {
```

Comment:
A theme here is making objects more immutable.  There were cases where fabrik8 was writing to the object passed in as args and messing up further tests.